x/mongo/driver/topology/server_test.go

```diff
@@ -457,17 +455,18 @@ func TestServer(t *testing.T) {
 			// On connection failure, the connection is removed and closed after delivering the
 			// error to Connection(), so it may still count toward the generation connection count
 			// briefly. Wait up to 100ms for the generation connection count to reach the target.
+			generation, _ := server.pool.generation.getGeneration(&serviceID)
 			assert.Eventuallyf(t,
 				func() bool {
-					generation := server.pool.generation.getGeneration(&serviceID)
+					generation, _ := server.pool.generation.getGeneration(&serviceID)
 					numConns := server.pool.generation.getNumConns(&serviceID)
 					return generation == wantGeneration && numConns == wantNumConns
 				},
 				100*time.Millisecond,
 				1*time.Millisecond,
 				"expected generation number %v, got %v; expected connection count %v, got %v",
 				wantGeneration,
-				server.pool.generation.getGeneration(&serviceID),
+				generation,
```

Comment:
Pre-fetching the generation number will lead to confusing test failure messages because we expect it to be updated concurrently while `assert.Eventuallyf` is running. We want the error message to show what the generation is when the assertion fails.

A possibly simpler approach is to log the state of the system when the assertion fails. For example:
```go
assert.Eventuallyf(t,
	func() bool {
		generation := server.pool.generation.getGeneration(&serviceID)
		generation, _ := server.pool.generation.getGeneration(&serviceID)
		numConns := server.pool.generation.getNumConns(&serviceID)

		match := generation == wantGeneration && numConns == wantNumConns
		if !match {
			t.Logf("Waiting for generation number %v, got %v", wantGeneration, generation)
			t.Logf("Waiting for connection count %v, got %v", wantNumConns, numConns)
		}
		return match
	},
	100*time.Millisecond,
	10*time.Millisecond,
	"expected generation number and connection count never matched")
```
Note that the example also changes the check interval from 1ms to 10ms to reduce log noise in the case there is a failure.